x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/history/SnapshotHistoryStore.java

```diff
@@ -59,85 +56,29 @@ public void putAsync(SnapshotHistoryItem item) {
                 SLM_HISTORY_INDEX_ENABLED_SETTING.getKey(), item);
             return;
         }
-        logger.trace("about to index snapshot history item in index [{}]: [{}]", SLM_HISTORY_ALIAS, item);
-        ensureHistoryIndex(client, clusterService.state(), ActionListener.wrap(createdIndex -> {
-            try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
-                item.toXContent(builder, ToXContent.EMPTY_PARAMS);
-                IndexRequest request = new IndexRequest(SLM_HISTORY_ALIAS)
-                    .source(builder);
-                client.index(request, ActionListener.wrap(indexResponse -> {
-                    logger.debug("successfully indexed snapshot history item with id [{}] in index [{}]: [{}]",
-                        indexResponse.getId(), SLM_HISTORY_ALIAS, item);
-                }, exception -> {
-                    logger.error(new ParameterizedMessage("failed to index snapshot history item in index [{}]: [{}]",
-                        SLM_HISTORY_ALIAS, item), exception);
-                }));
-            } catch (IOException exception) {
+        logger.trace("about to index snapshot history item in index [{}]: [{}]", SLM_HISTORY_DATA_STREAM, item);
+        Metadata metadata = clusterService.state().getMetadata();
+        if (metadata.dataStreams().containsKey(SLM_HISTORY_DATA_STREAM) == false &&
+            metadata.templatesV2().containsKey(SLM_TEMPLATE_NAME) == false) {
+            logger.error(new ParameterizedMessage("failed to index snapshot history item in index [{}]: [{}]",
```

Comment:
I think this should be a bit more descriptive about why it failed (such as mentioning that the data stream and template were missing) since we don't have an exception in this case

Reply:
You are right, I reworded error message